packages/@aws-cdk/aws-logs/lib/log-retention.ts

```diff
@@ -93,6 +99,8 @@ export class LogRetention extends Construct {
           base: retryOptions.base?.toMilliseconds(),
         } : undefined,
         RetentionInDays: props.retention === RetentionDays.INFINITE ? undefined : props.retention,
+        //only add RemovalPolicy if it is set to DESTROY
+        RemovalPolicy: props.removalPolicy && props.removalPolicy === cdk.RemovalPolicy.DESTROY ? props.removalPolicy : undefined,
```

Comment:
There's no reason here to not just pass `props.removalPolicy` here. 
```suggestion
        RemovalPolicy: props.removalPolicy,
```

Reply:
When I wrote this line I thought only add RemovalPolicy if it is set to DESTROY(not add it if it is undefined or set RETAIN), so it is more reasonable to set it as it is?

Reply:
You're later checking against `RemovalPolicy.DESTROY`, which kind of does make it so that you don't need `RemovalPolicy.RETAIN` but there's no reason to add this extra logic here. It's better to just pass the value as input by the user. You're not getting any benefit from the extra logic and it opens up the possibility of more failure points when more logic is introduced.